websocket/src/emscripten_callbacks.cpp

```diff
@@ -21,6 +21,8 @@ EM_BOOL Emscripten_WebSocketOnError(int eventType, const EmscriptenWebSocketErro
 EM_BOOL Emscripten_WebSocketOnClose(int eventType, const EmscriptenWebSocketCloseEvent *websocketEvent, void *userData) {
     DebugLog(1, "WebSocket OnClose");
     WebsocketConnection* conn = (WebsocketConnection*)userData;
+    // if the client disconnected through a call to websocket.disconnect() we
+    // receive this callback after the connection has been destroyed
```

Comment:
As I mentioned in the other discussion, I suggest adding another state to the connection (e.g. CLOSING), so you can wait until you've got the "MESSAGE_TYPE_CLOSE" message.